x-pack/plugins/cloud_security_posture/public/pages/rules/rules_container.tsx

```diff
@@ -93,6 +94,43 @@ export const RulesContainer = () => {
     params.benchmarkVersion
   );
 
+  const rulesStates = useCspGetRulesStates();
+  const arrayRulesStates = Object.values(rulesStates.data || {});
+  const filteredRulesStates: any[] = arrayRulesStates.filter(
+    (e: any) =>
+      e.benchmark_id === params.benchmarkId && e.benchmark_version === 'v' + params.benchmarkVersion
+  );
+
+  const allNewRulesItems: CspBenchmarkRulesWithStatus[] = useMemo(() => {
+    if (data === undefined) return [];
+    return data?.items.reduce((res: any[], obj1) => {
+      const rulesKey =
+        obj1.metadata.benchmark.id +
+        ';' +
+        obj1.metadata.benchmark.version +
+        ';' +
+        obj1.metadata.benchmark.rule_number;
+      const match = rulesStates?.data?.[rulesKey];
+      if (match) {
+        res.push({
+          ...obj1,
+          ...{ status: rulesStates?.data?.[rulesKey].muted === true ? 'muted' : 'unmuted' },
+        });
+      } else {
+        res.push({ ...obj1, ...{ status: 'unmuted' } });
+      }
+      return res;
+    }, []);
+  }, [data, rulesStates?.data]);
```

Comment:
```
const rulesWithStatus: CspBenchmarkRulesWithStatus[] = useMemo(() => {
  if (!data) return [];

  return data.items.map(obj1 => {
    const rulesKey =
      `${obj1.metadata.benchmark.id};${obj1.metadata.benchmark.version};${obj1.metadata.benchmark.rule_number}`;
    
    const match = rulesStates?.data?.[rulesKey];
    const status = match?.muted ? 'muted' : 'unmuted';

    return { ...obj1, status: status || 'unmuted' };
  });
}, [data, rulesStates?.data]);
```

i suggest trying this simpler version, there is no need to use reduce here as far as i could tell